packages/query/src/query-engine/index.ts

```diff
@@ -1,105 +1,36 @@
 import {
-	ApolloClient,
-	ApolloError,
-	InMemoryCache,
-	type NormalizedCacheObject,
-	type ServerError,
-	type ServerParseError,
-} from '@apollo/client';
-import {
-	Logger,
 	type EnqueuedScriptProps,
 	type GlobalHeadProps,
 	type ScriptModuleProps,
 	type StyleSheetProps,
 } from '@snapwp/core';
-import { getConfig, getGraphqlUrl } from '@snapwp/core/config';
-
+import { getConfig } from '@snapwp/core/config';
 import {
 	GetCurrentTemplateDocument,
 	GetGeneralSettingsDocument,
 	GetGlobalStylesDocument,
 } from '@graphqlTypes/graphql';
-
 import parseGeneralSettings from '@/utils/parse-general-settings';
 import parseGlobalStyles from '@/utils/parse-global-styles';
 import parseTemplate from '@/utils/parse-template';
-
 import type { BlockData } from '@snapwp/types';
 
 /**
  * Singleton class to handle GraphQL queries using Apollo.
  */
 export class QueryEngine {
-	private static instance: QueryEngine | null = null;
-	private static graphqlEndpoint: string;
-	private static homeUrl: string;
-	private static apolloClient: ApolloClient< NormalizedCacheObject >;
-
-	private static isClientInitialized = false;
-
-	/**
-	 * Initializer.
-	 */
-	public static initialize(): void {
-		QueryEngine.graphqlEndpoint = getGraphqlUrl();
-
-		const { wpHomeUrl } = getConfig();
-		QueryEngine.homeUrl = wpHomeUrl;
-
-		QueryEngine.apolloClient = new ApolloClient( {
-			uri: QueryEngine.graphqlEndpoint,
-			cache: new InMemoryCache(),
-		} );
-	}
-
-	/**
-	 * Returns the singleton instance of QueryEngine.
-	 * @throws Throws error if instance is not initialized with config.
-	 *
-	 * @return The QueryEngine instance.
-	 */
-	public static getInstance(): QueryEngine {
-		if ( ! QueryEngine.instance ) {
-			QueryEngine.instance = new QueryEngine();
-		}
-		return QueryEngine.instance;
-	}
-
 	/**
 	 * Fetches global styles.
-	 *
 	 * @return The template data fetched for the uri.
 	 */
 	static getGlobalStyles = async (): Promise< GlobalHeadProps > => {
-		if ( ! QueryEngine.isClientInitialized ) {
-			QueryEngine.initialize();
-		}
-
-		try {
-			const data = await QueryEngine.apolloClient.query( {
-				query: GetGlobalStylesDocument,
-				fetchPolicy: 'network-only', // @todo figure out a caching strategy, instead of always fetching from network
-				errorPolicy: 'all',
-			} );
-
-			return parseGlobalStyles( data );
-		} catch ( error ) {
-			if ( error instanceof ApolloError ) {
-				logApolloErrors( error );
-
-				// If there are networkError throw the error with proper message.
-				if ( error.networkError ) {
-					// Throw the error with proper message.
-					throw new Error(
-						getNetworkErrorMessage( error.networkError )
-					);
-				}
-			}
+		const data = await getConfig().queryEngine.fetchQuery( {
```

Comment:
Am I mistaken or in react do we always want to be object destructuring `getConfig()` versus direct access?

cc @ayushnirwal per https://github.com/rtCamp/snapwp/pull/144#discussion_r2032485169

Reply:
People generally lean towards destructuring. cc: @SH4LIN